service/src/main/kotlin/fi/espoo/evaka/assistanceneed/decision/AssistanceNeedDecisionController.kt

```diff
@@ -29,13 +32,29 @@ class AssistanceNeedDecisionController(
         db: Database,
         user: AuthenticatedUser,
         @PathVariable childId: ChildId,
-        @RequestBody body: PostAssistanceNeedDecisionRequest
+        @RequestBody body: AssistanceNeedDecisionRequest
     ): AssistanceNeedDecision {
         Audit.ChildAssistanceNeedDecisionCreate.log(targetId = childId)
         accessControl.requirePermissionFor(user, Action.Child.CREATE_ASSISTANCE_NEED_DECISION, childId)
         return db.connect { dbc ->
             dbc.transaction { tx ->
-                tx.insertAssistanceNeedDecision(childId, body.decision)
+                var decision: AssistanceNeedDecision = body.decision
+                if (decision.guardianInfo.isEmpty()) {
+                    val guardianIds = tx.getChildGuardians(childId)
+                    decision = body.decision.copy(
+                        guardianInfo = guardianIds.map { gid ->
+                            val guardian = tx.getPersonById(gid)
+                            AssistanceNeedDecisionGuardian(
+                                personId = gid,
+                                name = "${guardian?.lastName} ${guardian?.firstName}",
```

Comment:
Is it necessary to get the guardian info as it is only used for the name here, and this decision is only used for creating a new one? The name specified here won't be saved to the database anyways, so maybe it is unnecessary to do `tx.getPersonById` overall?